### DISCUSSION TITLE
What are the usages and behaviors of `VisitAll` and `VisitMutAll`?

### DISCUSSION THREAD

**Discussion by hyf0** [swc-project/swc#6457] [Q&A] [answered]
I look into the code, but it's hard to understand the code behind many macros. I did see one test using the `VisitAll` trait,  which isn't of help to understanding.

**Comment by kdy1** [comment, marked as answer]
It's used to visit every node without `n.visit_children_with(self)` in each visitor method.

**Reply by kdy1** [reply]
It's not used by swc, and it's simply a public api
